src/Accelerators/NNPA/Dialect/ZHigh/ZHighOps/Conv2D/Conv2D.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===------------------ Conv2D.cpp - ZHigh Operations ---------------------===//
+//
+// Copyright 2019-2022 The IBM Research Authors.
+//
+// =============================================================================
+//
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Accelerators/NNPA/Dialect/ZHigh/ZHighOps/ShapeHelper.hpp"
+
+using namespace mlir;
+using namespace onnx_mlir;
+
+namespace onnx_mlir {
+namespace zhigh {
+
+//===----------------------------------------------------------------------===//
+// ShapeHelper
+//===----------------------------------------------------------------------===//
+
+LogicalResult ZHighConv2DOpShapeHelper::computeShape() {
+  ZHighConv2DOp convOp = llvm::dyn_cast<ZHighConv2DOp>(op);
+  ZHighConv2DOp::Adaptor operandAdaptor(operands);
+  // Get operands.
+  // X: [B, HI, WI, CI]
+  Value X = operandAdaptor.input();
+  // W: [KH, KW, CI, CO]
+  Value W = operandAdaptor.input_kernel();
+  // Get attributes.
+  ArrayAttr strides = convOp.strides();
+  StringRef paddingType = convOp.padding_type();
+
+  // Get bounds
+  MemRefBoundsIndexCapture XBounds(X);
```

Comment:
I know this is a bit of a pain, but I would like to discontinue the "Capture" calls, as they don't really abide by the new scheme. You can look into the new code that I wrote, 
```C++
  MemRefBoundsIndexCapture XBounds(X);
  IndexExpr B = XBounds.getDim(0);
```
should be replaced by
```C++
  createIE->getShapeAsDim(X, 0)
```
which will automatically generate the right code for analysis & lowering as it relies on IndexExprBuilders to do the right thing. In addition, I think it is more readable because the new function name tells exactly what is done.

It's a bit of work, sorry about that. I have changed all occurrence of the old interface in the new methods that I migrated, now there are still a few remaining for ops that did not use shape inference in lowering.. I will work on these, it would be much appreciated if you could work on the accel ones. Thanks

Reply:
I replaced all `MemRefBoundsIndexCapture` by `getShapeAsDims` as you suggested. Now, there is no `MemRefBoundsIndexCapture` in NNPA code. Thanks!